File: food_delivery/src/main/java/com/datn/food_delivery/models/FavoriteFood.java
package com.datn.food_delivery.models;

public class FavoriteFood {
    private Long favoriteFood_id;
    private Long product_id;
    private String email;

    public FavoriteFood() {
    }

    public FavoriteFood(Long favoriteFood_id, Long product_id, String email) {
        this.favoriteFood_id = favoriteFood_id;
        this.product_id = product_id;
        this.email = email;
    }

    public FavoriteFood(Long product_id) {
        this.product_id = product_id;
    }

    public long getFavoriteFood_id() {
        return favoriteFood_id;
    }

    public void setFavoriteFood_id(long favoriteFood_id) {
        this.favoriteFood_id = favoriteFood_id;
    }

    public Long getProduct_id() {
        return product_id;
    }

    public void setProduct_id(Long product_id) {
        this.product_id = product_id;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }
}

File: food_delivery_flutter/lib/Model/FavoriteFood.dart

class FavoriteFood{
  final int favoriteFood_id;
  final int product_id;
  final String? email;

  FavoriteFood({required this.favoriteFood_id, required this.product_id, required this.email});

  factory FavoriteFood.fromJson(Map<String, dynamic> json) {
    return FavoriteFood(
        favoriteFood_id: json['favoriteFood_id'],
        product_id: json['product_id'],
        email: json['email'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'favoriteFood_id': favoriteFood_id,
      'product_id': product_id,
      'email': email,
    };
  }

  factory FavoriteFood.toJson(Map<String, dynamic> json) {
    return FavoriteFood(
        favoriteFood_id: json['favoriteFood_id'],
        product_id: json['product_id'],
        email: json['email'],
    );
  }
}